### PR TITLE
[greyhoundSideCar] exposed stopConsumer endpoint

### PR DESCRIPTION
**Pull request by itakedem**
replaced old creators with new one

### CODE REVIEW

File: greyhoundsidecar/src/main/scala/greyhound/SidecarService.scala [hunk 1]
```diff
@@ -85,46 +86,72 @@ class SidecarService(tenantRegistry: TenantRegistry,
...
 
+  override def stopConsuming(request: StopConsumingRequest): IO[Status, StopConsumingResponse] = {
+    for {
+      _ <- validateConsumersExist(request.consumersDetails, request.registrationId)
+      _ <- stopConsuming0(request)
+    } yield StopConsumingResponse()
+  }
+
+  private def stopConsuming0(request: StopConsumingRequest): ZIO[Any, Status, Seq[Unit]] = {
+    ZIO.foreachPar(request.consumersDetails) { consumerDetails =>
```
Comment (natansil) [inline greyhoundsidecar/src/main/scala/greyhound/SidecarService.scala:121]: Isn't `ZIO.foreachPar` cool :)

**Review by natansil — COMMENTED** [review]
Great looking PR.
Great Work!
please see a few comments (and gushing about ZIO :) )

File: greyhoundsidecar/src/test/scala/greyhound/MultiTenantTest.scala [hunk 2]
```diff
@@ -104,7 +104,7 @@ object MultiTenantTest extends JUnitRunnableSpec with KafkaTestSupport with Conn
             consumers = Seq(Consumer(contextForTopic.consumerId, contextForTopic.group, contextForTopic.topicName))))
           result <- sidecarService.startConsuming(StartConsumingRequest(
             registrationId = user1TenantId,
-            consumers = Seq(Consumer(contextForTopic.consumerId, contextForTopic.group, contextForTopic.topicName)))).either
+            consumers = Seq(Consumer(contextForTopic.consumerId, contextForTopic.group, contextForTopic.topicName)))).delay(1.second).either
```
[the remark this answers is not in the document]
Comment (avigdor-mans) [reply greyhoundsidecar/src/test/scala/greyhound/MultiTenantTest.scala:107]: it started after we change the create consumer from being an Object to a Class and we are concern about it. Maybe you have some thought about it.

[the remark this answers is not in the document]
Comment (natansil) [reply greyhoundsidecar/src/test/scala/greyhound/MultiTenantTest.scala:107]: > from being an Object to a Class 

Not likely this is the cause... but something else in the PR changed timing...
anyway please check flakiness with `@@ nonFlaky`

**Pull request merged**